### DISCUSSION TITLE
[Feature Request] NG Alert allow data source variable in alert definition

### DISCUSSION THREAD

**Discussion by d-shi** [grafana/grafana#37570] [Alerting]
The new NG Alerting is really cool. I had originally thought that it would allow template variables in alert query definitions, but it does not seem to. I did look over all the open issues mentioning NG Alert, but did not see this called out.

We currently have a prometheus instance for each of our clusters, but a single grafana instance to store dashboards and alert definitions. Each cluster has its own prometheus data source. Right now this means we have to duplicate all the alerts for each cluster, which is very cumbersome and error prone. I was hoping with NG Alerting we could resolve this by using a dashboard variable to find all prometheus data sources (let's call the var `cluster`), and then in the alert query we could use `${cluster}` as the data source. This is possible in a regular graph panel, but not for alerts.

Will this be a possible feature in the future? Or is the stance that we just have a single giant prometheus that scrapes all clusters, or use federation? (or just switch to using alertmanager)

**Comment by d-shi** [comment]
I ended up solving this by implementing thanos so we have a single prometheus compatible endpoint that can fan out and get metrics from all the separate prometheus instances running in all our clusters.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
